# This is the patch from tigervnc-1.0.0/unix/xserver16.patch
# modified to apply to the xserver 1.7.x sources used in our builds

# Only changes needed to make the patch apply should happen in
# this patch - other bug fixes & customizations should go into
# separate patches, so they aren't lost when this patch is
# replaced with new versions from upstream.

diff -urp -x '*~' -x '*.orig' unix/xserver/configure.ac unix/xserver/configure.ac
--- unix/xserver/configure.ac	2010-01-07 18:27:12.367254728 -0800
+++ unix/xserver/configure.ac	2010-01-07 18:27:15.429140493 -0800
@@ -30,7 +30,6 @@ AC_INIT([xorg-server], 1.7.4, [https://b
 RELEASE_DATE="2010-01-08"
 AC_CONFIG_SRCDIR([Makefile.am])
 AM_INIT_AUTOMAKE([dist-bzip2 foreign])
-AM_MAINTAINER_MODE
 
 AC_CONFIG_FILES([
 	shave
@@ -64,6 +63,7 @@ dnl forcing an entire recompile.x
 AC_CONFIG_HEADERS(include/version-config.h)
 
 AC_PROG_CC
+AC_PROG_CXX
 AM_PROG_AS
 AC_PROG_INSTALL
 AC_PROG_LN_S
@@ -1401,6 +1401,9 @@ if test "x$XVFB" = xyes; then
 	AC_SUBST([XVFB_SYS_LIBS])
 fi
 
+dnl Xvnc DDX
+AC_SUBST([XVNC_CPPFLAGS], ["-DHAVE_DIX_CONFIG_H $XEXT_INC $FB_INC $MI_INC $RENDER_INC $RANDR_INC"])
+AC_SUBST([XVNC_LIBS], ["$MAIN_LIB $FB_LIB $FIXES_LIB $XEXT_LIB $CONFIG_LIB $DBE_LIB $RECORD_LIB $GLX_LIBS $RANDR_LIB $RENDER_LIB $DAMAGE_LIB $MIEXT_DAMAGE_LIB $MIEXT_SHADOW_LIB $XI_LIB $XKB_LIB $XKB_STUB_LIB $COMPOSITE_LIB"])
 
 dnl Xnest DDX
 
@@ -1441,6 +1444,8 @@ xorg_bus_linuxpci=no
 xorg_bus_bsdpci=no
 xorg_bus_sparc=no
 
+AC_DEFINE_UNQUOTED(XORG_VERSION_CURRENT, [$VENDOR_RELEASE], [Current Xorg version])
+
 if test "x$XORG" = xyes; then
 	XORG_DDXINCS='-I$(top_srcdir)/hw/xfree86 -I$(top_srcdir)/hw/xfree86/include -I$(top_srcdir)/hw/xfree86/common'
 	XORG_OSINCS='-I$(top_srcdir)/hw/xfree86/os-support -I$(top_srcdir)/hw/xfree86/os-support/bus -I$(top_srcdir)/os'
@@ -1683,7 +1688,6 @@ if test "x$XORG" = xyes; then
 	AC_DEFINE(XORGSERVER, 1, [Building Xorg server])
 	AC_DEFINE(XFree86Server, 1, [Building XFree86 server])
 	AC_DEFINE(XFree86LOADER, 1, [Building loadable XFree86 server])
-	AC_DEFINE_UNQUOTED(XORG_VERSION_CURRENT, [$VENDOR_RELEASE], [Current Xorg version])
 	AC_DEFINE(NEED_XF86_TYPES, 1, [Need XFree86 typedefs])
 	AC_DEFINE(NEED_XF86_PROTOTYPES, 1, [Need XFree86 helper functions])
 	AC_DEFINE(__XSERVERNAME__, "Xorg", [Name of X server])
@@ -2129,6 +2133,7 @@ hw/dmx/input/Makefile
 hw/dmx/glxProxy/Makefile
 hw/dmx/Makefile
 hw/vfb/Makefile
+hw/vnc/Makefile
 hw/xnest/Makefile
 hw/xwin/Makefile
 hw/xquartz/Makefile
diff -urp -x '*~' -x '*.orig' unix/xserver/hw/Makefile.am unix/xserver/hw/Makefile.am
--- unix/xserver/hw/Makefile.am	2009-01-07 15:14:46.000000000 -0800
+++ unix/xserver/hw/Makefile.am	2010-01-07 18:27:15.429299239 -0800
@@ -33,7 +33,8 @@ SUBDIRS =			\
 	$(XNEST_SUBDIRS)	\
 	$(DMX_SUBDIRS)		\
 	$(KDRIVE_SUBDIRS)	\
-	$(XQUARTZ_SUBDIRS)
+	$(XQUARTZ_SUBDIRS)	\
+	vnc
 
 DIST_SUBDIRS = dmx xfree86 vfb xnest xwin xquartz kdrive
 
diff -urp -x '*~' -x '*.orig' unix/xserver/mi/miinitext.c unix/xserver/mi/miinitext.c
--- unix/xserver/mi/miinitext.c	2010-01-07 18:27:12.377142821 -0800
+++ unix/xserver/mi/miinitext.c	2010-01-07 18:27:15.429639241 -0800
@@ -290,6 +290,7 @@ extern void GEExtensionInit(INITARGS);
 #ifdef SolarisIAExtension
 extern void IAExtensionInit(void);
 #endif
+extern void vncExtensionInit(INITARGS);
 
 /* The following is only a small first step towards run-time
  * configurable extensions.
@@ -476,6 +477,7 @@ InitExtensions(int argc, char *argv[])
 #ifdef XF86BIGFONT
     if (!noXFree86BigfontExtension) XFree86BigfontExtensionInit();
 #endif
+    vncExtensionInit();
 #if !defined(NO_HW_ONLY_EXTS)
 #if defined(XF86VIDMODE)
     if (!noXFree86VidModeExtension) XFree86VidModeExtensionInit();
